CONTRIBUTING.md

```diff
@@ -255,143 +285,83 @@ cuda-memcheck python <program_name>.py <program_arguments>
-setting the `-G` compile option locally in your `cpp/CMakeLists.txt` for that file.
-Here is an example of adding the `-G` option to the compile command for
-`src/copying/copy.cu` source file:
+Therefore, it is recommended to add device debug symbols only to specific files by setting the `-G`
+compile option locally in your `cpp/CMakeLists.txt` for that file. Here is an example of adding the
+`-G` option to the compile command for `src/copying/copy.cu` source file:
 
 ```
 set_source_files_properties(src/copying/copy.cu PROPERTIES COMPILE_OPTIONS "-G")
 ```
 
-This will add the device debug symbols for this object file in libcudf.so.
-You can then use `cuda-dbg` to debug into the kernels in that source file.
-
-### Building and Testing on a gpuCI image locally
-
-Before submitting a pull request, you can do a local build and test on your machine that mimics our gpuCI environment using the `ci/local/build.sh` script.
-For detailed information on usage of this script, see [here](ci/local/README.md).
-
+This will add the device debug symbols for this object file in `libcudf.so`.  You can then use
+`cuda-dbg` to debug into the kernels in that source file.
 
-## Automated Build in Docker Container
+## Code Formatting
 
-A Dockerfile is provided with a preconfigured conda environment for building and installing cuDF from source based off of the main branch.
+### Python
 
-### Prerequisites
+cuDF uses [Black](https://black.readthedocs.io/en/stable/), [isort](https://pycqa.github.io/isort/),
+and [flake8](https://flake8.pycqa.org/en/latest/) to ensure a consistent code format throughout the
+project. They have been installed during the `cudf_dev` environment creation.
 
-* Install [nvidia-docker2](https://github.com/nvidia/nvidia-docker/wiki/Installation-(version-2.0)) for Docker + GPU support
-* Verify NVIDIA driver is `450.80.02` or higher
-* Ensure CUDA 11.0+ is installed
-
-### Usage
-
-From cudf project root run the following, to build with defaults:
-```bash
-docker build --tag cudf .
-```
-After the container is built run the container:
-```bash
-docker run --runtime=nvidia -it cudf bash
-```
-Activate the conda environment `cudf` to use the newly built cuDF and libcudf libraries:
-```
-root@3f689ba9c842:/# source activate cudf
-(cudf) root@3f689ba9c842:/# python -c "import cudf"
-(cudf) root@3f689ba9c842:/#
-```
-
-### Customizing the Build
-
-Several build arguments are available to customize the build process of the
-container. These are specified by using the Docker [build-arg](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg)
-flag. Below is a list of the available arguments and their purpose:
-
-| Build Argument | Default Value | Other Value(s) | Purpose |
-| --- | --- | --- | --- |
-| `CUDA_VERSION` | 11.0 | 11.2.2 | set CUDA version |
-| `LINUX_VERSION` | ubuntu18.04 | ubuntu20.04 | set Ubuntu version |
-| `CC` & `CXX` | 9 | 10 | set gcc/g++ version |
```

Comment:
Most of the times as a python developer, I run into gcc compilation issues. And when I do this is the place I would look at does my env have the right compatible gcc versions, since we are removing this what is the alternative place we can look at?

Reply:
This list was supposed to be Docker parameters but Docker builds were removed in #10069. This document lists compiler support above for `gcc 9.3+`. (However, that's not entirely true because we don't have compatibility with gcc 11 yet...)

We could try to expand the list of requirements here to mention Python versions, but this document has a tendency to become outdated very quickly. We only updated the Python package specifiers in #10579 (April 4, after [dropping Python 3.7 on February 3](https://docs.rapids.ai/notices/rsn0014/)), so I'm hesitant to add another source of truth here.

I would vote to take no action on this for now. I think this is better suited for a broader overhaul of both the C++/Python developer docs as we've proposed for some time now.